### DISCUSSION TITLE
check for local ./pypi/whl's listed in all.json

### DISCUSSION THREAD

**Discussion by laynr** [jupyterlite/jupyterlite#556] [Q&A] [answered]
My ./pypi/all.json is 3.5MB and points to local `*-py3-none-any.whl` files also in the local  pypi directory. Because it is local, I had hoped in a pyodide notebook to be able to do:

```
import micropip
await micropip.install('tenacity')
await micropip.install('plotly')
```
However when the cell is ran, the browser console shows a request to `https://pypi.org/pypi/tenacity/json` **Is it possible/intended to check  local before doing an external request?**

I know explicitly pointing it to the local whl works:
```
import micropip
await micropip.install('https://<localserver>/public/pypi/tenacity-8.0.1-py3-none-any.whl)
await micropip.install('https://<localserver>/public/pypi/plotly-5.6.0-py2.py3-none-any.whl)
```

Jupyterlite was built with `jupter lite build --pyodide ./pyodide-build-0.19.1tar.bz2 --output-dir ./public` with tenacity and plotly in the pypi folder.

Also, are there any plans to auto install dependencies (so  that when micropip.install('plotly') is called tenacity is first installed)?

**Comment by laynr** [comment, marked as answer]
ah hello `piplite`!

```
import piplite
await piplite.install(['pandas', 'plotly'])
```

Will leave by question here in case it helps others.

**Reply by psychemedia** [reply]
Hmm... I tried [building a wheel in a GitHub action](https://github.com/ouseful-PR/jupyter-splitview/blob/main/.github/workflows/jupyterlite-publish.yml), which [seems to get published okay](https://github.com/ouseful-PR/jupyter-splitview/tree/gh-pages/pypi) but if I try to `micropip` or `piplite` install it, the wheel is still pulled from pypi rather than locally. Did I miss a build setting somewhere?

**Comment by jtpio** [comment]
> Will leave by question here in case it helps others.

Thanks! Maybe we should make it more visible in the docs too.
